
# CPGTA - Clinical ProteoGenomics Tumor Analysis
## Overview

The CPGTA package is designed for processing and analyzing clinical cancer-related ProteoGenomics data, including multi-omics data download, correlation analysis, differential analysis, immunosubtype classification, and survival analysis. The package supports various types of omics data such as transcriptomics, proteomics, and phosphoproteomics.


## Installation

You can install the latest version of the `CPGTA` package from GitHub using the `devtools` package:

```{r cars}
# Install devtools if not already installed
install.packages("devtools")

# Install cptca from GitHub
devtools::install_github("LYING7797/CPGTA")
```

## Required Data
The package requires additional data files to function properly. Please download the required data using one of the following methods:

From Zenodo (Recommended):
Download the data from our Zenodo repository (DOI: 10.5281/zenodo.17357650)

After downloading, place the "data1" folder in the installation directory of the CPGTA package. The typical location would be within your R library path where the CPGTA package is installed.

To find the correct location, you can use:

```{r cars}
find.package("CPGTA")
```
Place all the data in the directory shown by the command above.


## Functions


cptod()
This function is used to obtain cancer multi-omics data from the data directory. You can filter and download relevant data according to parameters such as cancer type, data category, data type, and sample type.

Arguments:
@param cancer.type String. The name or abbreviation of the cancer type.
(valid range: 
cancer_type                              abbreviation
Ovarian Serous Cystadenocarcinoma        OV        
Colon Adenocarcinoma                     COAD      
Breast Invasive Carcinoma                BRCA      
Uterine Corpus Endometrial Carcinoma     UCEC      
Clear Cell Renal Cell Carcinoma          CCRCC     
Lung Adenocarcinoma                      LUAD      
Pediatric Brain Tumors                   Brain Cancer
Hepatocellular Carcinoma                 LIHC      
Glioblastoma                             GBM       
Early Onset Gastric Cancer               Gastric Cancer
Head and Neck Squamous Cell Carcinoma    HNSCC     
Lung Squamous Cell Carcinoma             LUSC      
Pancreatic Ductal Adenocarcinoma         PDAC      
Non-Clear Cell Renal Cell Carcinoma      Non-ccRCC 
Oral Squamous Cell Carcinoma             OSCC )
 
@param data.category String. 
(valid range:
Biospecimen
Clinical data
Copy Number Variation
Simple Nucleotide Variation
Transcriptome
Proteome
Phosphoproteome).

@param data.type String. Data type, usually "Normalized"("nor") or "Tidied"("tidy"). For transcriptome data,
  it can be "counts". Default is NULL, meaning no restriction on data type.
@param sample.type String or string vector. Sample type, such as "tumor" or "normal".
  Default is NULL, meaning no restriction on sample type.
@param PDC.study.identifier String. Specific PDC study identifier (e.g., "PDC000125").
  Default is NULL, meaning all relevant PDC studies will be found according to cancer type.
@return Returns a list containing data files filtered by the parameters.

example:
```{r}
# example
clinical_data <- cptod(
  cancer.type = "BRCA",
  data.category = "Clinical data"
)


all_rna_data <- cptod(
  cancer.type = "LUAD",
  data.category = "Transcriptome"
)

snv_data <- cptod(
  cancer.type = "COAD",  
  data.category = "Simple Nucleotide Variation"
)

result1 <- cptod(
  cancer.type = "OV",
  data.category = "Biospecimen",
  PDC.study.identifier = "PDC000113",
  data.type = NULL,
  sample.type = NULL)


result2 <- cptod(
  cancer.type = "OV",
  data.category = "Transcriptome",
  PDC.study.identifier = NULL,
  data.type = "Normalized",
  sample.type = "tumor")

```


cptca()
This function calculates the gene-wise correlation between two omics datasets (e.g., Transcriptome and Proteome) for a specified cancer type and gene. It reads, merges, and aligns data from multiple PDC studies if necessary,and visualizes the correlation distribution between tumor and normal samples, highlighting the specified gene.

Arguments:
@param gene.name String. The gene name of interest for correlation analysis.
@param cancer.type String. Cancer type name or abbreviation.
@param data.category Character vector of length 2. Specifies the two omics to compare, e.g., c("Transcriptome", "Proteome").

@return Returns a list containing:
   \item{correlation_data}{A data frame of correlation values for all genes in tumor and normal samples.}
   \item{plot}{A ggplot object visualizing the correlation distributions and highlighting the specified gene.}

example:
```{r}
res <- cptca(
  gene.name    = "TP53",
  cancer.type  = "OV",
  data.category = c("Transcriptome", "Proteome"))
```

cptda()
Perform differential analysis between tumor and normal samples for a specific cancer type using RNA-seq data. The function reads counts data from tumor and normal samples based on PDC codes, performs differential expression analysis using DESeq2, and generates volcano plots highlighting significant genes based on specified thresholds.

@param cancer.type String. The name or abbreviation of the cancer type.
@param log2FC Numeric. Threshold for log2 fold change, up to one decimal place.
@param top.gene Integer. Number of top differentially expressed genes to label in the volcano plot.
@param gene.name Character vector or NULL. Optional. Names of specific genes to highlight in the volcano plot. If provided and the genes exist in the dataset, they will be highlighted with a different color and size regardless of whether they're in the top genes.
@return A list of lists, with each sublist containing:
   \item{diff_table}{A data frame of differential analysis results, including gene names, log2FC, p-values, adjusted p-values, and group assignment.}
   \item{volcano_plot}{A ggplot2 object of the volcano plot with labeled top genes and optional highlighted gene.}


example:
```{r}

res <- cptda (
  cancer.type  = "UCEC",
  log2FC       = 1.5,  
  top.gene     = 10,
  gene.name = "TP53"
)


```

cptic()
This function classifies tumor samples based on their immunological profiles,facilitating the study and understanding of the tumor microenvironment. It integrates clinical and proteogenomic data to assign immunosubtypes to tumor samples for a specified cancer type and omics data category.

@param cancer.type String. The full name or abbreviation of the cancer type (e.g., "BRCA" or "Breast Cancer").
@param data.category String. Omics data category; must be either "Transcriptome" or "Proteome".

@return A data frame containing classified tumor samples and their immunological subtypes
        for the specified cancer type and omics category.

example:
```{r}
result_rna <- cptic(cancer.type = "LUSC", data.category = "Transcriptome")
print(result_rna)

result_pro <- cptic(cancer.type = "GBM", data.category = "Proteome")
print(result_pro)

```


cptsa()
This function performs survival analysis by correlating clinical outcomes with molecular profiles
for a specified cancer type, omics data category, and gene of interest. It generates survival curves to visualize the relationship between gene expression and patient prognosis, helping to identify potential biomarkers for patient stratification.

@param cancer.type String. The full name or abbreviation of the cancer type (e.g., "BRCA" or "Breast Cancer").
@param data.category String. Omics data category; must be either "Transcriptome" or "Proteome".
@param gene.name String. The gene symbol or name to be analyzed in the survival analysis.

@return A list containing:
  \item{km_fit}{A 'survfit' object representing the Kaplan-Meier survival fit.}
  \item{logrank_test}{A log-rank test result comparing survival between groups.}
  \item{plot}{A 'ggsurvplot' object visualizing the survival curves and statistics.}

example:
```{r}
res <- cptsa(
  cancer.type  = "OV",
  data.category = "Proteome",
  gene.name = "TP53"
  )
```

cptsnv()
This function enables the analysis of somatic single-nucleotide variations (SNVs) in cancer. It supports the identification of key mutations by integrating data from multiple PDC identifiers or cancer types. The function generates comprehensive visualizations, such as mutation frequency plots, oncoplots, and heatmaps, to help uncover potential mutation-based biomarkers for patient stratification and therapeutic insights.

@param cancer.type String (optional). The full name or abbreviation of the cancer type (e.g., "BRCA" or "Breast Cancer").
@param PDC.study.identifier String or vector (optional). The PDC study identifier(s) to be analyzed.
@param top_n_genes Integer. Number of top mutated genes to display in summary plots and co-occurrence analysis. Default is 20.
@param min_mut_freq Numeric. Minimum mutation frequency threshold for gene selection. Default is 0.05.

@return A list containing analysis results, summary statistics, and generated visualization objects.

example:
```{r}
res <- cptsnv(
  cancer.type = "LUAD",
  PDC.study.identifier = NULL,
  top_n_genes = 20,
  min_mut_freq = 0.05
  )
```

cptgev()
This function creates violin plots to compare gene expression levels between tumor and normal tissues
across multiple cancer types . It supports visualization of both transcriptomic and proteomic data.

@param gene_name Character string specifying the gene symbol to analyze (e.g., "BRCA1", "TP53").
@param data_category Character string indicating the data type to analyze, must be either "Proteome" or "Transcriptome".

@return A list containing two elements:
  \item{plot}{A ggplot object with violin plots showing gene expression in tumor vs. normal tissues across cancer types}
  \item{data}{A data frame containing the processed expression data used for plotting}
```{r}
result <- cptgev(
  gene_name = "BRCA1",
  data_category = "Transcriptome"
)
```






License
This package is licensed under the MIT license. See the LICENSE file for details.






